src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -542,19 +542,50 @@ private SourceEventSymbol GetEventSymbol(EventDeclarationSyntax eventDeclaration
 
             private Symbol GetMemberSymbol(string memberName, TextSpan memberSpan, NamedTypeSymbol container, SymbolKind kind)
             {
-                // return container.GetMembers(methodSyntax.Identifier.ValueText).OfType<SourceMethodSymbol>().Single(m => m.Locations.Any(l => l.SourceTree == tree && methodSyntax.Span.Contains(l.SourceSpan)));
-                foreach (Symbol sym in container.GetMembers(memberName))
+                Debug.Assert(kind is SymbolKind.Method or SymbolKind.Property or SymbolKind.Event);
+
+                if (container is SourceMemberContainerTypeSymbol { PrimaryConstructor: not null } sourceMemberContainerTypeSymbol)
+                {
+                    foreach (Symbol sym in sourceMemberContainerTypeSymbol.GetMembersToMatchAgainstDeclarationSpan())
```

Comment:
>sourceMemberContainerTypeSymbol.GetMembersToMatchAgainstDeclarationSpan() [](http://example.com/codeflow?start=43&length=73)

Why use `GetMembersToMatchAgainstDeclarationSpan()` rather than using `GetMembers(memberName)`? Is it to avoid cycles?

Reply:
> Why use ```GetMembersToMatchAgainstDeclarationSpan()``` rather than using GetMembers(memberName)? Is it to avoid cycles?

Yes, ```GetMembers``` includes backing fields, but we need binders to figure them out - cycle.